src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EmbeddedDataContainerNode.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.TypeSystem;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// Represents a node that contains a set of embedded objects. The main function is 
+    /// to serve as a base class, providing symbol name boundaries and node ordering.
+    /// </summary>
+    public abstract class EmbeddedDataContainerNode : ObjectNode
+    {
+        private ObjectAndOffsetSymbolNode _startSymbol;
+        private ObjectAndOffsetSymbolNode _endSymbol;
+
+        public ObjectAndOffsetSymbolNode StartSymbol => _startSymbol;
+        public ObjectAndOffsetSymbolNode EndSymbol => _endSymbol;
+
+        protected EmbeddedDataContainerNode(string startSymbolMangledName, string endSymbolMangledName)
+        {
+            _startSymbol = new ObjectAndOffsetSymbolNode(this, 0, startSymbolMangledName, true);
+            _endSymbol = new ObjectAndOffsetSymbolNode(this, 0, endSymbolMangledName, true);
+        }
+
+        protected override int ClassCode => (int)ObjectNodeOrder.EmbeddedDataContainerNode;
+
+        protected override int CompareToImpl(ObjectNode other, TypeSystemComparer comparer, NameMangler nameMangler)
+        {
+            return StartSymbol.GetMangledName(nameMangler).CompareTo(((EmbeddedDataContainerNode)other).StartSymbol.GetMangledName(nameMangler));
```

Comment:
Using mangled names for comparison is kind of a cop-out. They are rather expensive to compute and sorting will ask for them 2*N*log(N) times. I would rather see CompareToImpl not even give a chance to use them in the method sig, if we can help it.